custom_components/google_home/config_flow.py

```diff
@@ -51,28 +51,58 @@ async def async_step_user(
             return self.async_abort(reason="single_instance_allowed")
 
         if user_input is not None:
-            username = user_input[CONF_USERNAME]
-            password = user_input[CONF_PASSWORD]
             session = async_create_clientsession(self.hass)
-
-            if len(password) < MAX_PASSWORD_LENGTH:
+            try:
+                username = user_input[CONF_USERNAME]
+                password = user_input[CONF_PASSWORD]
+            except KeyError:
+                # either no username or password provided
+                username = None
+                password = None
+            
+            try:
+                master_token = user_input[CONF_MASTER_TOKEN]
+            except KeyError:
+                # no master token provided
+                master_token = None
```

Comment:
this works, but if you want to simplify, i would suggest going for
```suggestion
  username = user_input.get(CONF_USERNAME)
  password = user_input.get(CONF_PASSWORD)
  master_token = user_input.get(CONF_MASTER_TOKEN)
```

Reply:
Thanks. Didn't know about the get method :D